src/server/command_registry.cc

```diff
@@ -191,35 +204,66 @@ CommandRegistry& CommandRegistry::operator<<(CommandId cmd) {
 
   absl::InlinedVector<std::string_view, 2> maybe_subcommand = StrSplit(cmd.name(), " ");
   const bool is_sub_command = maybe_subcommand.size() == 2;
-  if (const auto it = cmd_rename_map_.find(maybe_subcommand.front()); it != cmd_rename_map_.end()) {
+  auto it = cmd_rename_map_.find(maybe_subcommand.front());
+  if (it != cmd_rename_map_.end()) {
     if (it->second.empty()) {
       return *this;  // Incase of empty string we want to remove the command from registry.
     }
     k = is_sub_command ? absl::StrCat(it->second, " ", maybe_subcommand[1]) : it->second;
   }
 
+  std::optional<CommandId> cloned = std::nullopt;
+  for (const auto& [alias, target] : alias_map_) {
```

Comment:
what's the reason for doing `O(n)` lookup here?

Reply:
I thought of using a `command -> alias` map originally which will be `O(1)` but if user aliases a command to multiple keywords, which seems a valid use case, then only the last alias will be picked up. If we assume and assert that one command is aliased only once then we can have a constant time lookup. 

Otherwise we can add another map with the reverse lookup or use a map with `command -> vector of aliases`. I will look into fixing this.

Reply:
Actually the current behavior is incorrect anyway. If we have multiple aliases mapped to one command, then in the lookup we only process the first alias and then break. So this won't process subsequent mappings. I will change this.

Reply:
Changed it to map command to alias, only one alias per command is allowed. The lookup will be constant time now.